libs/community/langchain_community/document_loaders/parsers/audio.py

```diff
@@ -12,6 +12,144 @@
+            Literal["json", "text", "srt", "verbose_json", "vtt"], None
+        ] = None,
+        temperature: Union[float, None] = None,
+    ):
+        self.api_key = api_key
+        self.base_url = base_url or os.environ.get("AZURE_OPENAI_ENDPOINT")
+        self.api_version = api_version or os.environ.get("OPENAI_API_VERSION")
+
+        self.deployment_id = deployment_id
+        self.chunk_duration_threshold = chunk_duration_threshold
+        self.language = language
+        self.prompt = prompt
+        self.response_format = response_format
+        self.temperature = temperature
+
+    @property
+    def _create_params(self) -> Dict[str, Any]:
+        params = {
+            "language": self.language,
+            "prompt": self.prompt,
+            "response_format": self.response_format,
+            "temperature": self.temperature,
+        }
+        return {k: v for k, v in params.items() if v is not None}
+
+    def lazy_parse(self, blob: Blob) -> Iterator[Document]:
+        """Lazily parse the blob."""
+
+        import io
+
+        try:
+            import openai
+        except ImportError:
+            raise ImportError(
+                "openai package not found, please install it with "
+                "`pip install openai`"
+            )
+        try:
+            from pydub import AudioSegment
+        except ImportError:
+            raise ImportError(
+                "pydub package not found, please install it with " "`pip install pydub`"
+            )
+
+        if is_openai_v1():
+            # api_key optional, defaults to `os.environ['AZURE_OPENAI_API_KEY']`
+            # api_version optional, defaults to `os.environ['OPENAI_API_VERSION']`
+            # azure_endpoint/base_rule optional,
+            # defaults to `os.environ['AZURE_OPENAI_ENDPOINT']`
+            client = openai.AzureOpenAI(
+                api_key=self.api_key,
+                azure_endpoint=self.base_url,
+                api_version=self.api_version,
+            )
+        else:
+            # Set the API key if provided
+            if self.api_key:
+                openai.api_key = self.api_key
+            if self.base_url:
+                openai.base_url = self.base_url
+
+        # Audio file from disk
+        audio = AudioSegment.from_file(blob.path)
+
+        # Define the duration of each chunk in minutes
+        # Need to meet 25MB size limit for Whisper API
+        chunk_duration = 20
+        chunk_duration_ms = chunk_duration * 60 * 1000
+
+        # Split the audio into chunk_duration_ms chunks
+        for split_number, i in enumerate(range(0, len(audio), chunk_duration_ms)):
+            # Audio chunk
+            chunk = audio[i : i + chunk_duration_ms]
+            # Skip chunks that are too short to transcribe
+            if chunk.duration_seconds <= self.chunk_duration_threshold:
+                continue
+            file_obj = io.BytesIO(chunk.export(format="mp3").read())
```

Comment:
Should we be hard coding this to mp3? Is format something that we should allow users to configure?
What will the behaviour be if we set format=mp3, when the data is encoded with something else?

Reply:
I've changed this section to us `os.path.splitext` to get the file extension from the file and use it to format the file_obj instead of hard code it.

I also added a file checking section to the code:
```
if not str(blob.path).endswith(self.input_format):
            raise ValueError(
                "File must be of one of the following types: "
                f"{str(self.input_format)}"
            )
```

`input_format` is a Tuple containing the allowed file types set by the Whisper API. I did this because the error returned when the file is the wrong type wasn't very informative and I thought this might work better. I also thought it might be good because the user could configure the `input_format` to only allow certain file types, if for whatever reason they wanted to only use say MP3 and MP4.

Let me know if this makes sense or if it should be removed.